### PR TITLE
Rename tools to agents

### PR DESCRIPTION
**Pull request by SimonRichardson**
Whilst updating some documentation in a previous PR (#12165) it was
noticed that this was using old terminology for agents. This PR cleans
up the naming of those naming conventions.

## QA Steps

```sh
juju metadata generate-agents -h
```

### CODE REVIEW

File: cmd/plugins/juju-metadata/validatetoolsmetadata.go [hunk 1]
```diff
@@ -44,12 +44,12 @@ type validateToolsMetadataCommand struct {
...
 }
 
-var validateToolsMetadataDoc = `
+var validateAgentsMetadataDoc = `
 validate-agents loads simplestreams metadata and validates the contents by
-looking for agent binaries belonging to the specified series, architecture, for the
-specified cloud. If version is specified, agent binaries matching the exact specified
-version are found. It is also possible to just specify the major (and optionally
-minor) version numbers to search for.
+looking for agent binaries belonging to the specified series, architecture,
```
Comment (achilleasa) [inline cmd/plugins/juju-metadata/validatetoolsmetadata.go:49]: s/series, architecture,/series and architecture/? (redundant commas)

File: cmd/plugins/juju-metadata/validatetoolsmetadata.go [hunk 2]
```diff
@@ -44,12 +44,12 @@ type validateToolsMetadataCommand struct {
...
-var validateToolsMetadataDoc = `
+var validateAgentsMetadataDoc = `
 validate-agents loads simplestreams metadata and validates the contents by
-looking for agent binaries belonging to the specified series, architecture, for the
-specified cloud. If version is specified, agent binaries matching the exact specified
-version are found. It is also possible to just specify the major (and optionally
-minor) version numbers to search for.
+looking for agent binaries belonging to the specified series, architecture, 
+for the specified cloud. If version is specified, agent binaries matching the 
+exact specified version are found. It is also possible to just specify the
```
Comment (achilleasa) [inline cmd/plugins/juju-metadata/validatetoolsmetadata.go:51]: Maybe `If version is specified, only agent binaries matching that exact version will be considered`?

File: cmd/plugins/juju-metadata/validatetoolsmetadata.go [hunk 3]
```diff
@@ -94,28 +95,27 @@ Examples:
...
+[ $RETVAL -eq 0 ] && echo Success || echo "Failure"
 `
 
-func (c *validateToolsMetadataCommand) Info() *cmd.Info {
+func (c *validateAgentsMetadataCommand) Info() *cmd.Info {
 	return jujucmd.Info(&cmd.Info{
 		Name:    "validate-agents",
-		Purpose: "validate agent metadata and ensure agent binary tarball(s) exist for Juju version(s)",
-		Doc:     validateToolsMetadataDoc,
+		Purpose: "validate agent metadata and ensure agent binary .tgz tarball(s) exist for Juju version(s)",
```
Comment (achilleasa) [inline cmd/plugins/juju-metadata/validatetoolsmetadata.go:112]:
```suggestion
		Purpose: "validate agent metadata and check that compressed tar archives (.tgz) with the Juju agent binaries are available",
```

I think there was a bit of duplication there:
.tgz = (compressed) tar archive
tarball = tar archive

**Review by achilleasa — APPROVED** [review]
Thanks for the text changes.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**